header-footer-grid/Core/Components/EddCartIcon.php

```diff
@@ -0,0 +1,217 @@
+
+
+/**
+ * Class Yoast_Breadcrumbs
+ *
+ * @package Neve_Pro\Modules\Header_Footer_Grid\Components
+ */
+class EddCartIcon extends Abstract_Component {
+
+	const COMPONENT_ID   = 'edd_cart';
+	const ICON_SELECTOR  = 'icon_selector';
+	const CART_LABEL     = 'cart_label';
+	const SIZE_ID        = 'icon_size';
+	const LABEL_SIZE_ID  = 'label_size';
+	const COLOR_ID       = 'color';
+	const HOVER_COLOR_ID = 'hover_color';
+
+
+	/**
+	 * Breadcrumbs constructor.
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 */
+	public function init() {
+		$this->set_property( 'label', __( 'Easy Digital Downloads Cart', 'neve' ) );
+		$this->set_property( 'id', self::COMPONENT_ID );
+		$this->set_property( 'width', 1 );
+		$this->set_property( 'default_selector', '.builder-item--' . $this->get_id() );
+	}
+
+	/**
+	 * Called to register component controls.
+	 *
+	 * @since   1.0.0
+	 * @access  public
+	 */
+	public function add_settings() {
+
+		do_action( 'nv_edd_cart_icon_component_controls' );
+
+		SettingsManager::get_instance()->add(
+			[
+				'id'                    => self::SIZE_ID,
+				'group'                 => $this->get_id(),
+				'tab'                   => SettingsManager::TAB_STYLE,
+				'transport'             => 'postMessage',
+				'sanitize_callback'     => 'absint',
+				'default'               => 15,
+				'label'                 => __( 'Icon Size', 'neve' ),
+				'type'                  => 'Neve\Customizer\Controls\React\Range',
+				'options'               => [
+					'input_attrs' => [
+						'min'        => 10,
+						'max'        => 100,
+						'defaultVal' => 15,
+					],
+				],
+				'live_refresh_selector' => $this->default_selector . ' span.nv-icon.nv-cart svg',
+				'live_refresh_css_prop' => [
+					'cssVar'  => [
+						'vars'     => '--iconSize',
+						'selector' => '.builder-item--' . $this->get_id(),
+						'suffix'   => 'px',
+					],
+					'type'    => 'svg-icon-size',
+					'default' => 15,
+				],
+				'section'               => $this->section,
+				'conditional_header'    => true,
+			]
+		);
+
+		SettingsManager::get_instance()->add(
+			[
+				'id'                    => self::COLOR_ID,
+				'group'                 => self::COMPONENT_ID,
+				'tab'                   => SettingsManager::TAB_STYLE,
+				'transport'             => 'postMessage',
+				'sanitize_callback'     => 'neve_sanitize_colors',
+				'label'                 => __( 'Color', 'neve' ),
+				'type'                  => 'neve_color_control',
+				'section'               => $this->section,
+				'live_refresh_selector' => true,
+				'live_refresh_css_prop' => [
+					'cssVar' => [
+						'vars'     => '--color',
+						'selector' => '.builder-item--' . $this->get_id(),
+					],
+					[
+						'selector' => $this->default_selector . ' svg',
+						'prop'     => 'fill',
+						'fallback' => 'inherit',
+					],
+					[
+						'selector' => $this->default_selector . ' .cart-icon-label',
+						'prop'     => 'color',
+						'fallback' => 'inherit',
+					],
```

Comment:
Are these needed if we use CSS variables? 🤔 

The fill and color properties should be set to the var in CSS and it should work fine. These were probably used for the old skin on other components (which we don't support anymore so we should be fine without them).

Reply:
@abaicus Thank you, I have tested without those keys and it seems to work fine. I had borrowed the code from our WooCommerce cart component implementation